### PR TITLE
533 run ci tests on fedora and macos

### PR DESCRIPTION
**Pull request by ErwanDemairy**
**PR description**

TO_BE_FILLED_BY_PR_CREATOR

**Developer Certificate Of Origin (DCO)**

By opening this merge request, you agree to the
[Developer Certificate of Origin (DCO)](https://github.com/fedbiomed/fedbiomed/blob/master/CONTRIBUTING.md#fed-biomed-developer-certificate-of-origin-dco)

This DCO essentially means that:

- you offer the changes under the same license agreement as the project, and
- you have the right to do that,
- you did not steal somebody else’s work.

**License**

Project code files should begin with these comment lines to help trace their origin:
```
# This file is originally part of Fed-BioMed
# SPDX-License-Identifier: Apache-2.0
```

Code files can be reused from another project with a compatible non-contaminating license.
They shall retain the original license and copyright mentions.
The `CREDIT.md` file and `credit/` directory shall be completed and updated accordingly.

**Guidelines for PR review**

General:

* give a glance to [DoD](https://fedbiomed.org/latest/developer/definition-of-done/)
* check [coding rules and coding style](https://fedbiomed.org/latest/developer/usage_and_tools/#coding-style)
* check docstrings (eg run `tests/docstrings/check_docstrings`)

Specific to some cases:

* update all conda envs consistently (`development` and `vpn`, Linux and MacOS)
* if modified researcher (eg new attributes in classes) check if breakpoint needs update (`breakpoint`/`load_breakpoint` in `Experiment()`, `save_state`/`load_state` in aggregators, strategies, secagg, etc.)
* if modified a component with versioning (config files, breakpoint, messaging protocol) then update the version following the rules in `common/utils/_versions.py`

### CODE REVIEW

**Comment by ErwanDemairy** [comment]
The PR is still in a "Draft" state; but keeping the "Draft" status prevents the github actions to be runned, which is the whole point of this PR.

File: .github/actions/unit-tests/action.yml [hunk 1]
```diff
@@ -19,14 +19,14 @@ runs:
       run: | 
         source ~/.bashrc
         conda clean -i -y
-        ./scripts/configure_conda researcher
+        ./scripts/configure_conda researcher node
```
Comment (mvesin) [inline .github/actions/unit-tests/action.yml:22]: `node` should not be installed in actions for unit tests, as unit tests don't use this environment (by convention it uses `researcher`)

If needed to MNIST test, its installation should be moved there.

Pushing a patch for this.

**Comment by mvesin** [comment]
hi @ErwanDemairy as discussed together:

- still need to document (public doc, attached to issue) the process for new slaves setup. This can be done after merging this PR.
 You can mimic what was done for Ubuntu slave: https://github.com/fedbiomed/fedbiomed/issues/531

- done updated the reference/backup private storage for credentials for accessing slaves

**Comment by mvesin** [comment]
@ErwanDemairy : FYI I also had to update:
https://github.com/fedbiomed/fedbiomed/settings/branches

- for both `develop` and `master`
- in "Require branches to be up to date before merging"

=> update the list of required tests to the current list of run tests (symptom: even with all tests succeeded, branch was not mergeable)

**Pull request merged**